### PR TITLE
Fix test that asserts originated member uuid

### PR DESCRIPTION
**Pull request by srknzl**
Updates client version to 5.0(should we do it? we should do for the test to run in master ci tests) and adds client version check of 4.2.1 to a test to fix back compatibility tests.

failing: https://github.com/srknzl/test/actions/runs/967207890
after tests update, the one that should pass: https://github.com/srknzl/test/actions/runs/967371138

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "hazelcast-client",
-    "version": "4.2.0",
+    "version": "5.0",
```
Comment (mdumandag) [inline package.json:3]: This should be `5.0.0` to follow the semantic versioning

File: test/integration/backward_compatible/sql/ExecuteTest.js [hunk 2]
```diff
@@ -499,6 +499,7 @@ describe('SqlExecuteTest', function () {
             client = await Client.newHazelcastClient({
                 clusterName: cluster.id
             });
+            TestUtil.markClientVersionAtLeast(this, '4.2.1');
```
Comment (mdumandag) [inline test/integration/backward_compatible/sql/ExecuteTest.js:502]: We haven't released the `4.2.1` client yet, and maybe we never will :smile: 

I think we could make this `5.0.0` and update it if we release the 4.2.1

Comment (mdumandag) [reply test/integration/backward_compatible/sql/ExecuteTest.js:502]: Also, can you put the comment you put below to this as well?

File: test/integration/backward_compatible/sql/ExecuteTest.js [hunk 3]
```diff
@@ -499,6 +499,10 @@ describe('SqlExecuteTest', function () {
             client = await Client.newHazelcastClient({
                 clusterName: cluster.id
             });
+            // There was a fix regarding originatingMemberId in 5.0 and 4.2.x after 4.2.
+            // TODO: Update here to be 4.2.1 if we release 4.2.1
+            // https://github.com/hazelcast/hazelcast-nodejs-client/pull/940
+            TestUtil.markClientVersionAtLeast(this, '5.0.0');
```
Comment (mdumandag) [inline test/integration/backward_compatible/sql/ExecuteTest.js:505]: Just a thought, what do you think about not skipping this test for 5.0- clients, but change the assertion according to the client version? So for 4.2 client, we could use member.uuid, for 5.0 we could use client uuid?

Comment (srknzl) [reply test/integration/backward_compatible/sql/ExecuteTest.js:505]: I thought of doing that, but I think the correct uuid is client uuid, isn't it? But at the same time, I think assertion of originatingMemberId is just a single part of this test and we should run the other parts for 4.2 clients.

In total, I think we can change the assertion as you suggested

Comment (mdumandag) [reply test/integration/backward_compatible/sql/ExecuteTest.js:505]: Yes, maybe don't check the originatingMemberUuid on 5.0- clients, but do other assertions on all versions

Comment (mdumandag) [reply test/integration/backward_compatible/sql/ExecuteTest.js]: Sorry, I didn't saw your last commit, I am fine with different assertions on different versions

Comment (srknzl) [reply test/integration/backward_compatible/sql/ExecuteTest.js]: Ok. Let's go with the current version in 78b0c3f

**Comment by srknzl** [comment]
passing test on 4.2.0 client against 4.2 server 

https://github.com/srknzl/test/runs/2903760274?check_suite_focus=true

**Review by mdumandag — APPROVED** [review]

File: test/integration/backward_compatible/sql/ExecuteTest.js [hunk 4]
```diff
@@ -538,7 +545,14 @@ describe('SqlExecuteTest', function () {
...
             error1.should.be.instanceof(getHazelcastSqlException());
             error1.code.should.be.eq(getSqlErrorCode().CONNECTION_PROBLEM);
-            error1.originatingMemberId.should.be.eq(client.connectionManager.getClientUuid());
+            // There was a fix regarding originatingMemberId in 5.0 and 4.2.x after 4.2.0.
+            // TODO: Update here to be 4.2.1 if we release 4.2.1
+            // https://github.com/hazelcast/hazelcast-nodejs-client/pull/940
+            if (TestUtil.isClientVersionAtLeast('5.0.0')) {
+                error1.originatingMemberId.should.be.eq(client.connectionManager.getClientUuid());
+            } else {
+                error1.originatingMemberId.should.be.eq(member.uuid);
```
Comment (srknzl) [inline test/integration/backward_compatible/sql/ExecuteTest.js:554]: should we need toString here? @mdumandag see https://github.com/srknzl/test/runs/2904025971?check_suite_focus=true

Comment (srknzl) [reply test/integration/backward_compatible/sql/ExecuteTest.js:554]: This is the updated version that passes:

https://github.com/srknzl/test/runs/2904280680?check_suite_focus=true

see tests branch in srknzl/test for my latest commit

**Pull request merged**